[2979][ADD] report_account_move_settlement

kanda999 opened the pull request:
[2979](https://www.quartile.co/web?#id=2979&action=771&model=project.task&view_type=form&menu_id=505)

kanda999 committed:
[ADD] report_commission_sst

kanda999 committed:
[ADD]report_account_move_settlement

yostashiro commented on report_account_move_settlement/__manifest__.py:
```diff
@@ -0,0 +1,23 @@
+    "license": "AGPL-3",
+    "depends": [
+        "account",
+        "product",
+        "commission",
```
```suggestion
```

yostashiro commented on report_account_move_settlement/data/report_paperformat_data.xml:
```diff
@@ -0,0 +1,16 @@
+<?xml version="1.0" encoding="utf-8" ?>
+<odoo>
+    <record id="paperformat_commission_settlement" model="report.paperformat">
+        <field name="name">Customer account Invoice report Format</field>
```
```suggestion
        <field name="name">Seller settlement report format</field>
```

yostashiro commented on report_account_move_settlement/data/report_paperformat_data.xml:
```diff
@@ -0,0 +1,16 @@
+        <field name="margin_right">7</field>
+        <field name="header_line" eval="False" />
+        <field name="header_spacing">20</field>
+        <field name="dpi">90</field>
```
```suggestion
        <field name="dpi">100</field>
```
To reduce the number of pages.

yostashiro commented on report_account_move_settlement/i18n/ja.po:
```diff
@@ -0,0 +1,128 @@
+
+#. module: report_account_move_settlement
+#: model_terms:ir.ui.view,arch_db:report_account_move_settlement.commission_settlement_main
+msgid "Bill"
+msgstr "請求"
```
```suggestion
msgid "Bill Content"
msgstr "請求内訳"
```
Update the originating string accordingly.

yostashiro commented on report_account_move_settlement/i18n/ja.po:
```diff
@@ -0,0 +1,128 @@
+
+#. module: report_account_move_settlement
+#: model_terms:ir.ui.view,arch_db:report_account_move_settlement.commission_settlement_main
+msgid "Items"
+msgstr "入札会項目"
```
```suggestion
msgid "Sold Items"
msgstr "落札済商品詳細"
```

yostashiro commented on report_account_move_settlement/models/account_move.py:
```diff
@@ -0,0 +1,18 @@
+    @api.depends("settlement_ids", "settlement_ids.event_ids")
+    def _compute_event_ids(self):
+        for move in self:
+            settlement = move.settlement_ids
+            move.event_ids = settlement.mapped("event_ids")
```
```suggestion
            move.event_ids = move.settlement_ids.mapped("event_ids")
```

yostashiro commented on report_account_move_settlement/readme/DESCRIPTION.rst:
```diff
@@ -0,0 +1,3 @@
+This module does the following:
+
+- This module adds custom invoice with settlement reports to account_move
```
```suggestion
- Adds settlement report print function to account.move

The added function is intended to be used by auction sellers so that they can keep
the details of the settlement invoice for their record.
```

yostashiro commented on report_account_move_settlement/reports/report_settlement_templates.xml:
```diff
@@ -0,0 +1,224 @@
+                        style="max-height: 45px;"
+                        alt="Logo"
+                    />
+                    <span>ASMM精算一覧書</span>
```
Avoid hardcoding a client specific title here.

Make it something like:
```.xml
<t t-if="company.seller_settlement_report_title">
    <t t-field="company.sellter_settlement_report_title" />
</t>
<t t-else="">
    <span>精算一覧書</span>
</t>
```

yostashiro commented on report_account_move_settlement/reports/report_settlement_templates.xml:
```diff
@@ -0,0 +1,224 @@
+                        </div>
+                        <div class="mt32" t-if="o.partner_id">
+                            <strong>Partner:</strong>
+                            <span t-field="o.partner_id" />
```
```suggestion
                            <span t-field="o.partner_id" t-options='{"widget": "contact", "fields": ["name"], "no_marker": True}' />
```

yostashiro commented on report_account_move_settlement/data/report_paperformat_data.xml:
```diff
@@ -0,0 +1,16 @@
+        <field name="orientation">Portrait</field>
+        <field name="margin_top">10</field>
+        <field name="margin_bottom">18</field>
+        <field name="margin_left">7</field>
+        <field name="margin_right">7</field>
```
Maybe the margins should be a bit larger.  Check with the client.

yostashiro commented on report_account_move_settlement/reports/report_settlement_templates.xml:
```diff
@@ -0,0 +1,224 @@
+                </div>
+                <div class="row" style="padding-top: 20px;">
+                    <div class="col-12">
+                        <div>精算書発行後の水曜日に会員様口座へ送金いたします。</div>
+                        <div
+                        >差し引き請求金額がマイナスの場合は右記口座まで会員様手数料負担にてお振込みをお願いいたします。</div>
```
Would be good to adjust the presented content based on `move_type`.

yostashiro reviewed (changes requested):
Interim feedback.

Please take care of the +/- presentation for credit notes.

It may be OK not to have the tax column in the invoice lines.  Let's go without it for now.

yostashiro commented:
@kanda999 I may update the code a bit and push shortly.

yostashiro committed:
[2979][IMP] base_custom_report, report_account_move_settlement: misc. updates

- Make bank_info text field
- Add settlement report comment fields to res.company
- Improve _compute_tax_totals_json_signed()
- Adjust report layout
